### PR TITLE
Markdown fixes

### PR DESCRIPTION
**Pull request by morags**
Added several languages and fixed a typo in `markdown.tmLanguage`.

### CODE REVIEW

**Comment by msftgits** [comment]
Hi, I am closing and re-opening this PR to bump the CLA bot. Sorry for the inconvenience!

File: extensions/markdown/syntaxes/markdown.tmLanguage [hunk 1]
```diff
@@ -2404,6 +2431,38 @@
 						</dict>
 					</array>
 				</dict>
+				<key>fenced_code_block_rust</key>
```
Comment (mjbvz) [inline extensions/markdown/syntaxes/markdown.tmLanguage]: Please update these to use the fenced code block logic that the other languages now use. It's rather horrific but it fixes a few important edge cases

**Comment by mjbvz** [comment]
Thank you for these improvements. They should be in the next insiders build

**Comment by morags** [comment]
Thank you.

**Pull request merged**